consensus/wal_test.go

```diff
@@ -60,3 +62,60 @@ func TestSearchForEndHeight(t *testing.T) {
 	assert.Equal(t, rs.Height, h+1, cmn.Fmt("wrong height"))
 
 }
+
+func init() {
```

Comment:
this will be run not only for tests, right? don't think we want this. is there some other way?

Reply:
It will only be run for tests, we could even throw in Java code in the `_test.go` and it won't be run until we do `go test`.

However, I'll make it clear with an update :)

Reply:
I've updated it with a sync.Once protection to initialize interface registration only once.

Reply:
Omg, I just asked. You didn't have to :)

Reply:
Hehe, that's alright, it was a great question. I've fixed it so that in the future there won't be any confusion :)